htlcswitch/link.go

```diff
@@ -2631,11 +2631,20 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 
 		heightNow := l.cfg.Switch.BestHeight()
 
-		fwdInfo := chanIterator.ForwardingInstructions()
+		fwdInfo, err := chanIterator.ForwardingInstructions()
+		if err != nil {
+			// TODO(roasbeef): fail back due to mal formed TLV
+			// stream?
```

Comment:
We don't yet have an error type to signal this to the caller. T-bast has proposed this type, but it hasn't yet made it all the way through the process. Once it has (and the rest of the related AMP changes have), we can follow back and remedy this, hence the placement of this TODO. 

If we don't handle it (as is now), then the caller won't get any feedback as to why their payment failed, and we won't forward/offer the HTLC to the next hop or the registry.

Reply:
Meaning that the channel will eventually force close?

I don't think we need to wait for t-bast's proposed type. It would be just for debugging anyway. Imo we should just return any of the invalid onion failures that we already have here.